### ISSUE TITLE
"Trigger failed" button is permanently disabled

### ISSUE THREAD

**Issue by jhainzlschmid** [ls1intum/Artemis#4025]
### Describe the bug

In the participations page of a programming exercise the "Trigger failed" button is permanently disabled no matter if the participations contain submissions, which had build failures.

### To Reproduce
[Here is the related testcase in TestQuality](https://ls1intum.testquality.com/project/10165/plan/17385/run/137794/result/1986463)

1. Create a course with a programming exercise and add multiple students to the course
2. Make a submission with all students, where atleast one submission should fail the build
3. Open the participations page of the exercise as instructor
4. See that the "Trigger failed" button is disabled even though atleast one participation failed the build

### Expected behavior

The "Trigger failed" button should be enabled and when clicking it all builds of participations, which failed the build, should trigger a rebuild on the CI service (Bamboo or Jenkins).

### Screenshots

The button is disabled:
![image](https://user-images.githubusercontent.com/12861668/134210515-661b8dd7-ac7a-40ba-85d6-12cdeac082df.png)
even though atleast one participation has failed the build:
![image](https://user-images.githubusercontent.com/12861668/134210636-1b22b449-b677-4c68-98e3-6176baef876e.png)

### What browsers are you seeing the problem on?

Chrome

### Additional context

I tested this on my local setup (jenkins + gitlab) and on TS3 (bamboo + bitbucket). I used the develop branch to test on.
The console does not show any errors.

**Comment by MaisiKoleni** [comment]
This is not a bug, at most a communication problem:
 - `Build Failed` as a result with a message (e.g. due to a compilation error) is no problem for Artemis because it is a valid result with feedback. Only the build of the submission failed, and we can work with that (0% score). Triggering the build of such a submission is actually *expected* to yield `Build Failed` again.
 - A submission that failed is one that has no result (exactly like in #3976) and the main problem is that there is no feedback and no basis for grading, a score for the latest submission is missing.

In short: `Build Failed` is a valid result that does not call for a rebuild, the button stays disabled. 

I would suggest closing this issue or rewriting it to address either:
1. A communication problem due to the UI in Artemis (but I don't see much of an issue here)
2. A communication problem due to the test description on ls1intum.testquality.com (but the test does mention this explicitly)
3. Missing Artemis documentation that would explain the difference

**Comment by jhainzlschmid** [comment]
This was a misunderstanding on my side. I retested it on ts3 and the functionality does indeed work. I made a submission as a student and stopped the build in the buildplan. After some time the page showed that one submission failed and the "Trigger failed" button was enabled. When clicking it the failed build does indeed rerun
